### DISCUSSION TITLE
Who's on First as a source of admin data?

### DISCUSSION THREAD

**Discussion by ianthetechie** [valhalla/valhalla#4031] [Q&A] [answered]
Out of curiosity, have there been any thoughts about WOF as an admin data source before? Maybe it's not quite as convenient(? not sure the right word) as processing a PBF file or three, but I think it'd *probably* tend to have fewer issues in most cases since it doesn't have the usual OSM issues with collaboratively editing huge relations. It also seems like it'd be a lot quicker and probably simpler to process.

Given that both Valhalla and WOF came out of Mapzen, I assume I'm missing some obvious reason why it's not a good fit, but I just had the thought and figured I'd ask.

**Comment by kevinkreiser** [comment]
we did indeed consider it before but there were 2 good reasons not to do it:

1. we already had an implementation in place before WOF existed
2. the more data dependencies you have the more difficult your system is to maintain
3. (edut) data in valhalla not matching what mappers did in osm can have unexpected consequences especially at edge cases but also its confusing to users why they don't match

at this point it might be just possible to generate the sqlite output we need from WOF with some simple ogr2ogr commands which does seem pretty low maintenance.

valhalla_build_admins could become a simple bash or python script potentially. I hesitate to actually endorse this idea without thinking about it more but it's certainly possible

**Comment by ImreSamu** [comment, marked as answer]
imho: 
In Europe, state borders often exhibit fascinating intricacies, and OpenStreetMap provides the unique benefit of synchronizing the road network with these borders. When either element is misaligned by a few meters, the street automatically shifts to the adjacent state. A prime example of this can be observed in the municipality of Rosbruck, France, where the state border is seamlessly aligned with specific roads. ( https://www.openstreetmap.org/way/75849477#map=17/49.15636/6.84721 )

another example:
* Gorizia (Italy) and Nova Gorica (Slovenia)  https://www.openstreetmap.org/way/292601333#map=19/45.95379/13.63486

**Reply by kevinkreiser** [reply]
yeah after i answered I had a similar thought. that people doing osm mapping would find it strange that the admin relations in valhalla don't match what they see in osm

**Comment by ianthetechie** [comment]
Leaving a comment in case anyone finds this by searching and add a third good reason not to do it: the WOF data is not just misaligned from roads, it's often rather low resolution compared to OSM and their hierarchies are seriously messed up in a number of areas (mostly capital cities, ironically; just look at Seoul or Tallinn for examples). I believe now more than ever that OSM is a better source of admin data, after doing some related projects over the past 6 months involving administrative boundaries.

**Reply by ImreSamu** [reply]
> . I believe now more than ever that OSM is a better source of admin data,

The other alternative is Overture Maps **"Administrative Boundaries (Admins) Theme:" ( ODBL )** 
* _"The sources of the admin theme are [OpenStreetMap](http://openstreetmap.org/) and Esri."_
* "Data in the Admin theme is licensed under [ODbL](https://opendatacommons.org/licenses/odbl/)."
*  https://overturemaps.org/overture-december-2023-release-notes/

**Reply by ianthetechie** [reply]
Yeah, I actually just left a similar comment on that over here: https://github.com/valhalla/valhalla/issues/4467#issuecomment-1868247627. I believe either Overture or Daylight are interesting as sources and intend to post findings on using those with Valhalla soon-ish.
